app/temporal/src/activities.ts

```diff
@@ -81,175 +88,150 @@ export function makeActivities(
             `Unsupported format ${params.format}`
           );
       }
-      const jsonWithRowIds: DataSet = json.map((row, index) => ({
-        __rowId: index,
+      const jsonWithRowIds: SourceDataSet = json.map((row, index) => ({
+        __sourceRowId: index,
         ...row,
       }));
-
-      const jsonData = Buffer.from(JSON.stringify(jsonWithRowIds));
-      await fileStore.putFile(
-        params.bucket,
-        params.outputFileReference,
-        jsonData
-      );
+      await database.mongoClient
```

Comment:
Wir sollten eventuell noch die Collection zuvor komplett leeren. Falls die Activity retried wird, haben wir sonst doppelt so viele Zeilen. Außerdem kann es sein, dass bei vielen Zeilen in insertMany failed (mongo splitted glaube ich bei 100k Rows)